File: fast_import.py
import os
import sys
import mysql.connector

HOST = "thomas.proxy.rlwy.net"
PORT = 45044
USER = "root"
PASSWORD = "qMsZwbiIngMfINmKygVSbMIiqJfdoTst"
DATABASE = "railway"
SQL_FILE = "C:/Users/Stephen Tiozon/Documents/GitHub/ASPLAN/osas_db_07_28_26_part2_ignore.sql"

def run_import():
    try:
        print("Connecting to DB...")
        connection = mysql.connector.connect(
            host=HOST,
            port=PORT,
            user=USER,
            password=PASSWORD,
            database=DATABASE
        )
        cursor = connection.cursor()
        print("Connected.")
        
        print("Reading SQL file...")
        with open(SQL_FILE, 'r', encoding='utf-8') as f:
            sql_content = f.read()
        print(f"Loaded {len(sql_content)} bytes. Executing...")
        
        # We can split by ; manually or just use multi=True
        # However, some SQL dumps contain DELIMITER statements or similar which break multi=True.
        # But this is just phpMyAdmin dump.
        
        results = cursor.execute(sql_content, multi=True)
        count = 0
        for res in results:
            count += 1
            if count % 100 == 0:
                print(f"Executed {count} statements...")
                
        connection.commit()
        cursor.close()
        connection.close()
        print(f"Done! Executed {count} statements.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    run_import()

File: fast_import.php
<?php
$_ENV['MYSQL_PUBLIC_URL'] = 'mysql://root:PIlezyGzBauvijKewcPUtNqUtETTNcfP@hayabusa.proxy.rlwy.net:58143/railway';
require_once 'config/database.php';

try {
    echo "Connecting to DB...\n";
    $conn = getDBConnection();
    echo "Connected.\n";
    
    $file = "osas_db_07_28_26_part2_ignore.sql";
    echo "Reading $file...\n";
    
    $content = file_get_contents($file);
    echo "Read " . strlen($content) . " bytes.\n";
    
    // The mysqli multi_query can execute the whole file!
    // But since we are using PDO fallback sometimes, it's better to split by statement
    // Or just use the underlying PDO directly
    if ($conn instanceof mysqli) {
        echo "Using MySQLi multi_query...\n";
        if ($conn->multi_query($content)) {
            do {
                if ($res = $conn->store_result()) {
                    $res->free();
                }
            } while ($conn->more_results() && $conn->next_result());
            echo "Successfully executed multi_query.\n";
        } else {
            echo "Error executing multi_query: " . $conn->error . "\n";
        }
    } else {
        echo "Using PDO...\n";
        $pdo = $conn->getPdo();
        $pdo->exec($content);
        echo "Successfully executed PDO exec.\n";
    }

} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}
